bootstrap/pkg/client/gcp/gcp.go

```diff
@@ -802,13 +790,87 @@ func (gcp *Gcp) createSecrets(options map[string]interface{}) error {
 		return fmt.Errorf("At least one of --%v or ENV `%v` needs to be set.",
 			string(kftypes.OAUTH_SECRET), CLIENT_SECRET)
 	}
-
-	return gcp.writeGcpSecret(k8sClient, KUBEFLOW_OAUTH, map[string][]byte{
+	return gcp.writeGcpSecret(client, KUBEFLOW_OAUTH, map[string][]byte{
 		strings.ToLower(CLIENT_ID):     []byte(oauthId),
 		strings.ToLower(CLIENT_SECRET): []byte(oauthSecret),
 	})
 }
 
+func (gcp *Gcp) createBasicAuthSecret(client *clientset.Clientset, options map[string]interface{}) error {
+	if !gcp.GcpApp.Spec.UseBasicAuth {
+		log.Infof("Not using basic auth, skip creating basic auth login secret.")
+		return nil
+	}
+	username := ""
+	if options[string(kftypes.BASIC_AUTH_USERNAME)] != nil &&
+		options[string(kftypes.BASIC_AUTH_USERNAME)].(string) != "" {
+		username = options[string(kftypes.BASIC_AUTH_USERNAME)].(string)
+	} else {
+		username = os.Getenv(BASIC_AUTH_USERNAME)
+	}
+	if username == "" {
+		return fmt.Errorf("At least one of --%v or ENV `%v` needs to be set.",
+			string(kftypes.BASIC_AUTH_USERNAME), BASIC_AUTH_USERNAME)
+	}
+
+	password := ""
+	if options[string(kftypes.BASIC_AUTH_PASSWORD)] != nil &&
+		options[string(kftypes.BASIC_AUTH_PASSWORD)].(string) != "" {
+		password = options[string(kftypes.BASIC_AUTH_PASSWORD)].(string)
```

Comment:
why do we have kftypes.BASIC_AUTH_PASSWORD and BASIC_AUTH_PASSWORD?

Reply:
it's part of `CliOptions` (string).

Reply:
I mean, why do we duplicate things at two places?

Reply:
was going to use ENV as well, thought twice, removed.